### PR TITLE
API /client/_suggestions should only return unconfigured clients

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

See title, context below.

We alse add a new index, see comment in the commit message below.

---

**Related issue or feature (if applicable):** https://discourse.pi-hole.net/t/entries-in-client-section-dont-get-removed-after-adding/73568/5

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Review by yubiuser — COMMENTED** [review]
Something seems to be missing still

**Comment by DL6ER** [comment]
Okay, I extended this to scanning both for "real" and "mock" (like `ip-1.2.3.4`) addresses, please try again. The query plan for this query is frighteningly complex but this isn't an often used endpoint, so that's what it is:
```
QUERY PLAN
|--SCAN n
|--LIST SUBQUERY 3
|  `--SCAN g.client USING COVERING INDEX sqlite_autoindex_client_1
|--LIST SUBQUERY 4
|  `--SCAN g.client USING COVERING INDEX sqlite_autoindex_client_1
|--CORRELATED SCALAR SUBQUERY 1
|  `--SEARCH na USING INDEX network_addresses_network_id_index (network_id=?)
|--CORRELATED SCALAR SUBQUERY 2
|  |--USE TEMP B-TREE FOR group_concat(DISTINCT)
|  `--SEARCH na USING INDEX network_addresses_network_id_index (network_id=?)
`--USE TEMP B-TREE FOR ORDER BY
```
At the very least, the generated byte code confirms that the two `WHERE` subqueries are indeed evaluated only once (at the beginning) and not for every row...

File: src/database/network-table.c [hunk 1]
```diff
@@ -203,6 +203,25 @@ bool create_network_addresses_with_names_table(sqlite3 *db)
...
+bool create_network_addresses_network_id_index(sqlite3 *db)
+{
+	// Return early if database is known to be broken
+	if(FTLDBerror())
+		return false;
+
+	// Create index on network_id column in network_addresses table
+	SQL_bool(db, "CREATE INDEX IF NOT EXISTS network_addresses_network_id_index ON network_addresses (network_id);");
+
+	// Update database version to 8
```
Comment (yubiuser) [inline src/database/network-table.c:215]:
```suggestion
	// Update database version to 20
```

**Review by yubiuser — APPROVED** [review]
Works fine now

**Review by yubiuser — APPROVED** [review]

**Pull request merged**